File: Top 50 Coding Interview Questions/25. Write a program for Binary to Decimal to conversion/BinarytoDecimalConversion.cpp
#include <iostream>
#include <cmath>
#include <string>

using namespace std;

// Function to convert binary to decimal
int binaryToDecimal(string binary) {
    int decimal = 0;
    int power = 0;

    // Traverse the binary string from the last character to the first
    for (int i = binary.length() - 1; i >= 0; i--) {
        // Convert character to integer (0 or 1)
        int bit = binary[i] - '0';

        // Add to the decimal result using the power of 2
        decimal += bit * pow(2, power);

        // Increment power for the next position
        power++;
    }
    return decimal;
}

int main() {
    string binary;

    // Input binary number as a string
    cout << "Enter a binary number: ";
    cin >> binary;

    // Call the conversion function
    int decimal = binaryToDecimal(binary);

    // Output the result
    cout << "Decimal equivalent: " << decimal << endl;

    return 0;
}

File: Top 50 Coding Interview Questions/25. Write a program for Binary to Decimal to conversion/BinarytoDecimalConversion.py
# Function to convert binary to decimal
def binary_to_decimal(binary):
    decimal = 0
    power = 0

    # Traverse the binary string from the last character to the first
    for bit in reversed(binary):
        # Add to the decimal result using the power of 2
        decimal += int(bit) * (2 ** power)
        power += 1

    return decimal

# Input binary number as a string
binary = input("Enter a binary number: ")

# Call the conversion function
decimal = binary_to_decimal(binary)

# Output the result
print("Decimal equivalent:", decimal)
